objects/src/accounts/account_id.rs

```diff
@@ -134,336 +129,471 @@ impl FromStr for AccountStorageMode {
     }
 }
 
+// ACCOUNT VERSION
+// ================================================================================================
+
+/// The version of an [`AccountId`].
+///
+/// Each version has a public associated constant, e.g. [`AccountVersion::VERSION_0`].
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+pub struct AccountVersion(u8);
```

Comment:
Yeah I think it makes sense.
I guess this leaves the option of changing only the ID with a version bump and _also_ the account, but not necessarily. So this might indeed be better than `AccountVersion`.